EIPS/eip-7530.md

```diff
@@ -0,0 +1,121 @@
+description: EVM Profiles for ZK Rollups
+author: Peter Robinson (@drinkcoffee)
+discussions-to: https://ethereum-magicians.org/t/eip-7530-evm-profiles-to-enable-distributed-sequencers-for-zk-rollups/15999
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-10-03
+---
+
+
+## Abstract
+
+This EIP defines a set of EVM Profiles. Each profile describes how the Ethereum Virtual Machine (EVM) should operate when configured with the profile. The profile specifies that some EVM opcodes and precompiles should operate in the usual way, whereas other opcodes and precompiles operate differently, for example reverting. 
+
+## Motivation
+
+Blockchains that use deterministic finality consensus protocols can be used as decentralized sequencers in zero knowledge rollup systems, as shown in the figure below. 
+
+![Decentralized Sequencer Architecture](../assets/eip-7530/architecture.png)
+
+
+The L2 blockchain acts as a decentralized sequencer, defining an ordered list of transactions, and the resulting state changes produced by executing those transactions. Finalised transactions and related state changes are consumed by zero knowledge provers. The provers generate proofs that show that executing the transactions results in the state changes. The proofs can be verified in verifier smart contracts on Ethereum.
+
+In the system described in the previous paragraph there are three Ethereum Virtual Machines: the EVM executing in the L2 blockchain clients and the EVM processor in the zk prover, and the EVM in the L1 blockchain (Ethereum). The EVM in the L2 blockchain clients and the EVM in the zk prover must operate in precisely the same way for the proof to be able to be verified in the verifier smart contract.
+
+Several "EVM compatible" zero knowledge proof systems are available at time of writing (October 2023). However, each prover has implemented a slightly different set of opcodes and precompiles, and has different behaviour for opcodes and precompiles that are not supported. 
+
+The purpose of this EIP is to define a standard set of opcodes and precompiles must be supported and the behaviour for unsupported opcodes and precompiles. The goal is to allow any blockchain client configured with this profile to be connected to any prover configured with this profile to create a workable zero knowledge rollup system with a decentralised sequencer.
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Hardfork Reference Point
+
+The baseline for the opcodes and precompiles defined in this specification is the Shanghai Hark Fork. All opcodes and precompiles not specified in the sections below MUST be fully supported.
+
+### EVM Profile A
+
+This profile describes a system that has minimal changes compared to Ethereum.
```

Comment:
could you create a table for the opcodes for each profile, that be more legible.

Reply:
I have create a Summary of Changes section that contains tables of changed opcodes and precompiles for each profile.